.travis.yml

```diff
@@ -0,0 +1,60 @@
+# This travis config:
+#   Invokes contain_build.sh to build npad-support using a docker build
+#   container.
+#   Handles gcloud credentials.
+#   On successful build (on pushes only) deploys rpm to a GCS bucket.
+#     * On sandbox-* branches, or forks, successful builds will push rpm to
+#       legacy-rpms-mlab-sandbox, to owner/repo/branch folder.
+#     * On untagged builds from m-lab repo dev branch, pushes rpm
+#       legacy-rpms-mlab-staging, to testing folder.
+#     * On tagged builds from m-lab repo, pushes rpm to
+#       legacy-rpms-mlab-staging, to staging folder.
+
+dist: trusty
+language: ruby
+services:
+- docker
+
+script:
+# TODO - change script to use :latest instead of :travis.
+# This script builds the rpm using the docker container, getting
+# the build params from the travis environment variables.
+- $TRAVIS_BUILD_DIR/travis/container_build.sh
+
+before_deploy:
+# If an encryption key exists, this will decrypt the service-account keys.
+# This may fail if the key has been overwritten since the file was
+# encoded.
+- $TRAVIS_BUILD_DIR/travis/decrypt.sh
+
+# TODO: Travis has GCS deployment support, but for now, we like
+# the security provided by using the service account.
+deploy:
+# PRIVATE: For development in forked repos, or branches starting with
+# sandbox-, successful build will attempt to push rpm to sandbox bucket.
+ - provider: script
+   on:
+     all_branches: true
+     condition: $TRAVIS_REPO_SLUG != m-lab/npad-support || $TRAVIS_BRANCH =~ sandbox-*
+   script: $TRAVIS_BUILD_DIR/travis/deploy.sh /tmp/legacy-rpm-writer.mlab-sandbox.json legacy-rpms-mlab-sandbox/$TRAVIS_REPO_SLUG/$TRAVIS_BRANCH
```

Comment:
When I look in the sandbox bucket I'm not sure which folders should be there and which are left over from earlier development. Would you be willing to delete the stale ones?

https://pantheon.corp.google.com/storage/browser/legacy-rpms-mlab-sandbox/?project=mlab-sandbox&organizationId=433637338589